Lib/test/test_tarfile.py

```diff
@@ -2283,6 +2283,18 @@ def test__all__(self):
             'SubsequentHeaderError', 'ExFileObject', 'main'}
         support.check__all__(self, tarfile, not_exported=not_exported)
 
+    def test_useful_error_message_when_modules_missing(self):
+        fname = os.path.join(os.path.dirname(__file__), 'testtar.tar.xz')
```

Comment:
Hummm, not a fan of having binary files checked into, even if they are only 172 bytes. Is possible to generate this one dynamically in a temporary directory?

Reply:
there are (at least) two other files by which I was taking precedence from:
- https://github.com/python/cpython/blob/master/Lib/test/testtar.tar
- https://github.com/python/cpython/blob/master/Lib/test/recursion.tar

it is possible to generate this file _sometimes_ -- though it requires lzma to be available.

I could also inline the bytestring, but I don't think that actually improves the situation

Reply:
Meh, too complex, let's go ahead and check them into.